client/lib/analytics/signup.js

```diff
@@ -81,6 +93,7 @@ export function recordSignupComplete(
 	// blog_id instead of site_id here. We keep using "siteId" otherwise since
 	// all the other fields still refer with "site". e.g. isNewSite
 	recordTracksEvent( 'calypso_signup_complete', {
+		elapsed_time: elapsedTime ?? getSignupCompleteElapsedTimeInSeconds(),
```

Comment:
The reason that `getSignupCompleteElapsedTimeInSeconds()` are conditionally called in two places is because `recordSignupComplete()` has an internal queue. If `now` flag is absent, what it does is to queue the function and then let the internal task consumer make the real call. Thus, the `elapsedTime ?? ...` above is for queuing the real elapsed time to the queue. That way when this function is called by the task consumer, `elapsedTime` here will be the correct elapsed time being queued previously. In the meantime, if `now` flag is enabled, this line will also guarantee that `getSignupCompleteElapsedTimeInSeconds()` is called since `elapsedTime` should be null in that case.

Reply:
Can't we just call `getSignupCompleteElapsedTimeInSeconds()` on line 64 and use that value. The closure will preserve the correct value. 

We do not want to add the `Queuing` time to the actual metric do we? since it's not relevant to the elapsed time of the UX ?

Reply:
Also do you think we can rename the prop to `time_since_signup_start` I have a feeling that we might need to add this time stamp across multiple events to make sense of the data and the above event name is a bit more discriptive.

 Actually I would prefer if we add this event prop to all other major signup events like `calypso_signup_actions_submit_step`, `calypso_new_user_site_creation`, `calypso_user_registration_complete` from the get go. But this can be followed up in another PR if you prefer 👍.

Reply:
I've updated the prop name as a slightly different `elapsed_time_since_start`, since I somehow like to keep `elapsed` and it's already under a signup event so I only include `start`. Does that make sense to you?

Also:
> Actually I would prefer if we add this event prop to all other major signup events like calypso_signup_actions_submit_step, calypso_new_user_site_creation, calypso_user_registration_complete from the get go. But this can be followed up in another PR if you prefer 

Yeah, let's do it as a follow-up when the need arise :)

Reply:
> Can't we just call getSignupCompleteElapsedTimeInSeconds() on line 64 and use that value. The closure will preserve the correct value.
> We do not want to add the Queuing time to the actual metric do we? since it's not relevant to the elapsed time of the UX ?

Good question and that's the tricky part. First of all, no, we don't want to add the queuing time to the actual metric. That's why `getSignupCompleteElapsedTime()` has to be called before the values go into the queue.

The way that "trigger queue" works is fairly simple(see [addToQueue()](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/analytics/queue.js#L66) and [runTrigger()](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/analytics/queue.js#L48) ). It pushes the function name and the arguments into the queue through `addToQueue()`, and then invoke them accordingly by `runTrigger()`. Thus, there 

Thus, let's say we put `getSignupCompletionTime()` call on Line 64, it'd actually be called twice: 
1. The 1st time: when `recordSignupComplete()` is called with `now` parameter as `null`
2. The 2nd time: when `recordSignupComplete()` is called again by `runTrigger()` 

It's fine, but I don't want it to be called twice like this. Does that make sense? :)